src/App.tsx

```diff
@@ -1,56 +1,78 @@
+interface Props {
+  delay: number,
+}
+
+function debounce(callback: (...args: string[]) => void, delay: number) {
+  let timerId = 0;
+
+  return (...args: string[]) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+export const App: React.FC<Props> = ({ delay }) => {
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    applyQuery(event.target.value);
+  };
+
+  const handleSelect = useCallback((currentPerson: Person) => {
+    setSelectedPerson(currentPerson);
+    setAppliedQuery('');
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer
+      .filter(person => person.name.toLowerCase()
+        .includes(appliedQuery.trim().toLowerCase()));
+  }, [appliedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson?.name} (${selectedPerson?.born} - ${selectedPerson?.died})` : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={classNames('dropdown',
+        {
+          'is-active': appliedQuery,
+        })}
+      >
         <div className="dropdown-trigger">
           <input
+            onChange={handleQueryChange}
             type="text"
             placeholder="Enter a part of the name"
             className="input"
           />
         </div>
 
-        <div className="dropdown-menu" role="menu">
-          <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
-        </div>
+        {appliedQuery && (
+          <PeopleList
+            people={filteredPeople}
+            onSelect={handleSelect}
+          />
+        )}
```

Comment:
You're already using functionality with classes (lines 56-60) for activating drop down list, so there is no need of using conditional rendering